static/app/actionCreators/pageFilters.tsx

```diff
@@ -409,7 +423,7 @@ async function persistPageFilters(filter: PinnedPageFilter | null, options?: Opt
  */
 async function updateDesyncedUrlState(router?: Router, shouldForceProject?: boolean) {
   // Cannot compare URL state without the router
-  if (!router) {
+  if (!router || !PageFiltersStore.shouldPersist) {
     return;
   }
```

Comment:
If the filters don't persist, then we shouldn't check for desync state, otherwise we'll get tons of false positives.